integration/https_test.go

```diff
@@ -751,114 +752,92 @@ func (s *HTTPSSuite) TestEntrypointHttpsRedirectAndPathModification(c *check.C)
 			sourceURL:   "http://127.0.0.1:8888/",
-			expectedURL: "https://test.com:8443/",
+			expectedURL: "https://<HOST>:8443/",
 		},
 		{
 			desc:        "Root Path with double trailing slash redirect",
-			host:        "test.com",
+			host:        []string{"test.com", "test2.com", "pow.com", "pow2.com"},
 			sourceURL:   "http://127.0.0.1:8888//",
-			expectedURL: "https://test.com:8443//",
+			expectedURL: "https://<HOST>:8443//",
 		},
 		{
-			desc:        "AddPrefix with redirect",
-			host:        "test.com",
+			desc:        "Path modify with redirect",
+			host:        []string{"test.com", "test2.com", "pow.com", "pow2.com"},
 			sourceURL:   "http://127.0.0.1:8888/wtf",
-			expectedURL: "https://test.com:8443/wtf",
+			expectedURL: "https://<HOST>:8443/wtf",
 		},
 		{
-			desc:        "AddPrefix with trailing slash redirect",
-			host:        "test.com",
+			desc:        "Path modify with trailing slash redirect",
+			host:        []string{"test.com", "test2.com", "pow.com", "pow2.com"},
 			sourceURL:   "http://127.0.0.1:8888/wtf/",
-			expectedURL: "https://test.com:8443/wtf/",
+			expectedURL: "https://<HOST>:8443/wtf/",
 		},
 		{
-			desc:        "AddPrefix with matching path segment redirect",
-			host:        "test.com",
+			desc:        "Path modify with matching path segment redirect",
+			host:        []string{"test.com", "test2.com", "pow.com", "pow2.com"},
 			sourceURL:   "http://127.0.0.1:8888/wtf/foo",
-			expectedURL: "https://test.com:8443/wtf/foo",
-		},
-		{
-			desc:        "Stripped URL Regex redirect",
-			host:        "foo.com",
-			sourceURL:   "http://127.0.0.1:8888/api",
-			expectedURL: "https://foo.com:8443/api",
-		},
-		{
-			desc:        "Stripped URL Regex with trailing slash redirect",
-			host:        "foo.com",
-			sourceURL:   "http://127.0.0.1:8888/api/",
-			expectedURL: "https://foo.com:8443/api/",
-		},
-		{
-			desc:        "Stripped URL Regex with path redirect",
-			host:        "foo.com",
-			sourceURL:   "http://127.0.0.1:8888/api/bacon",
-			expectedURL: "https://foo.com:8443/api/bacon",
-		},
-		{
-			desc:        "Stripped URL Regex with path and trailing slash redirect",
-			host:        "foo.com",
-			sourceURL:   "http://127.0.0.1:8888/api/bacon/",
-			expectedURL: "https://foo.com:8443/api/bacon/",
+			expectedURL: "https://<HOST>:8443/wtf/foo",
 		},
 	}
 
 	for _, test := range testCases {
-		test := test
+		for _, subtest := range test.host {
+			req, err := http.NewRequest("GET", test.sourceURL, nil)
+			c.Assert(err, checker.IsNil)
+			req.Host = subtest
 
-		req, err := http.NewRequest("GET", test.sourceURL, nil)
-		c.Assert(err, checker.IsNil)
-		req.Host = test.host
+			resp, err := client.Do(req)
+			c.Assert(err, checker.IsNil)
+			defer resp.Body.Close()
 
-		resp, err := client.Do(req)
-		c.Assert(err, checker.IsNil)
-		defer resp.Body.Close()
+			location := resp.Header.Get("Location")
+			expected := strings.Replace(test.expectedURL, "<HOST>", subtest, 1)
```

Comment:
You should maybe replace by 
```go
expected := fmt.Sprintf(test.expectedURL, host)
```

And replace all `<HOST>` occurrences in `expectedURL` by `%s`